### ISSUE TITLE
Error parsing MIDI input: Invalid system exclusive message

### ISSUE THREAD

**Issue by sevonj** [AlexCharlton/midi-msg#25]
Hi, I'm testing out this crate to maybe use it in [my midi player.](https://github.com/sevonj/sfontplayer)

I have quite a few midi files that result in a this parse error when opened.

[sample file](https://github.com/user-attachments/files/18135766/sample.zip) 
output:
```
Error parsing MIDI file at position 193: Error parsing MIDI input: Invalid system exclusive message
Encountered this error while parsing: track 0 event 4
The incomplete MidiFile that managed to be parsed: MidiFile { header: Header { format: MultiTrack, num_tracks: 22, division: TicksPerQuarterNote(120) }, tracks: [Midi([TrackEvent { delta_time: 0, event: Meta { msg: TrackName("Breaking the Law (Judas Priest)") }, beat_or_frame: 0.0 }, TrackEvent { delta_time: 0, event: Meta { msg: Copyright("Copyright � 1998 by Jean-Fr�d�ric Vachon (jfvachon@oricom.ca)") }, beat_or_frame: 0.0 }, TrackEvent { delta_time: 0, event: Meta { msg: Text("Words and Music by Glenn Tipton, Rob Halford and K.K. Downing\n") }, beat_or_frame: 0.0 }, TrackEvent { delta_time: 0, event: Meta { msg: Unknown { meta_type: 33, data: [0] } }, beat_or_frame: 0.0 }])] }

44242 bytes remain in the file. These are the next ones: [0, f0, a, 41, 10, 42, 12, 40, 0, 7f, 0, 41, f7, 0, ff, 21, 1, 0, 0, ff]
```

I haven't looked at all of them in detail, but at a glance it seems to me that it's midi-msg trying to read the message one byte early.

Opening the file in https://ide.kaitai.io/ shows the message parsed correctly.
![image](https://github.com/user-attachments/assets/2d448f12-3f5e-4a4e-8c7e-b1c2fa60ae1f)

**Comment by AlexCharlton** [comment]
Hi @sevonj, thanks for the report. There was indeed an off-by-one error that I fixed in https://github.com/AlexCharlton/midi-msg/commit/ebd0fbc287d869407e47e685ee7901ee6d431233

I'll publish a new patch release for this tomorrow.

**Comment by sevonj** [comment]
Thanks!

I built it from your latest commit, and can confirm that this fixed most of my midis.

Edit: Found another another one

[sample2.zip](https://github.com/user-attachments/files/18140860/sample2.zip)

```
56103 bytes remain in the file. These are the next ones: [7, 7b, 0, 8, 1, 0, 7, 40, 0, 8, b, 7f, 7, d3, 0, 8, e3, 0, 40, 0]
```
`7b` is the status byte.

**Comment by AlexCharlton** [comment]
The new error, as it happens, is not specific to midi files. It's a bug in the running channel mode handling. I missed a test case and had _another_ off-by-one error, where we were swallowing a byte we shouldn't have. Fixed in https://github.com/AlexCharlton/midi-msg/commit/18ba7da785cd6558ef4f398cd454bf6ec3ee9043

**Comment by AlexCharlton** [comment]
Published the new version v0.7.4 with both of these fixes. Thanks for reporting!

**Comment by sevonj** [comment]
Cool!

With the previous release, it seemed that almost half of my arbitrarily picked files failed. 
I tested 0.7.4 with a tool that goes through everything I have, and the result looks pretty nice.

```
Files:
OK:           1404
ERR:          8
ERR NotImpl:  16
Skipped:      0
```
